app/src/main/kotlin/compiler/ast/Program.kt

```diff
@@ -2,7 +2,7 @@ package compiler.ast
 
 data class Program(val globals: List<Global>) {
     sealed class Global {
-        data class VariableDefinition(val variable: Variable) : Global()
-        data class FunctionDefinition(val function: Function) : Global()
+        data class VariableDefinition(val variable: Variable) : Global(), AstNodeTypes.NamedNode
```

Comment:
Is it necessary for `NamedNode`s to point to `Variable` wrappers such as `Program.VariableDefinition` and `Statement.VariableDefinition`, and not just to `Variable`s itself? Same question for the `FunctionDefinition` interface - can't `VariableProperties` objects just point directly to `Function`s?